pandas/core/tools/timedeltas.py

```diff
@@ -25,10 +26,13 @@ def to_timedelta(arg, unit=None, errors="raise"):
     Parameters
     ----------
     arg : str, timedelta, list-like or Series
-        The data to be converted to timedelta.  The character M by itself,
-        e.g. '1M', is treated as minute, not month. The characters Y and y
-        are treated as the mean length of the Gregorian calendar year -
-        365.2425 days or 365 days 5 hours 49 minutes 12 seconds.
+        The data to be converted to timedelta.
+
+        .. deprecated:: 1.2
+            Strings denoting units with 'M', 'Y', 'm' or 'y' do not represent
+            unambiguous timedelta values durations and will removed in a future
+            version
+
     unit : str, optional
         Denotes the unit of the arg for numeric `arg`. Defaults to ``"ns"``.
```

Comment:
Off-topic:  parameter unit defaults to ``"ns"`` according to docstring, but the python code defaults it to `None`. Can I change it here or should there be a separate PR for this?

Reply:
We use this pattern commonly in pandas. Fine to leave as is.